semperfiwebdesign/all-in-one-seo-pack
# Don't run conflicting shortcodes in autogenerated descriptions

Issue #1747

## Proposed changes

This fixes an issue where error WooCommerce can't display errors on the "My Account" page when the "Run shortcodes in autogenerated descriptions" setting is enabled in AIOSEOP.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Improves existing functionality
- Improves existing code

## Checklist


## Testing instructions

Enable "autogenerated descriptions" and "run shortcodes in autogenerated descriptions" in the General Settings menu, then navigate to the "My Account" page that WooCommerce generates and enter no or incorrect login details. WooCommerce should now be able to display error notifications again.

## Further comments

Specifically, this PR adds a new function (aioseop_contains_conflicting_shortcodes) in aioseop_functions.php where we can add more shortcodes in the future that are known to conflict with this setting.

Also adds a unit test for this new function.
